File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        while (l < r) {
            int middle = l + (r - l) / 2;
            if (nums[middle] >= target) r = middle;
            else l = middle + 1;
        }
        return (l < nums.size() && nums[l] == target) ? l : -1;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        l = 0 
        r = 0
        res = 0

        while r < len(nums)-1:
            maxMfJump = 0
            print("l: ", l, "r: ", r)
            for i in range(l, r + 1) :
                maxMfJump = max(maxMfJump, i+nums[i])
                print('maxMfJump', maxMfJump)
            l = r + 1
            r = maxMfJump
            res += 1
            print('res: ', res, '\n#######################')

        return res
        